### ISSUE TITLE
Facehuggers go permanently UNCONSCIOUS on failed leaps

### ISSUE THREAD

**Issue by Rohesie** [tgstation/TerraGov-Marine-Corps#364]
## Reproduction:

Whenever a facehugger leaps without being thrown  and fails to hit a target, it will go UNCONSCIOUS and then stop aging, trying to leap again, or be able to be attached manually. The only solution is to put it back to the egg and remove it.

The problem is in `Facehuggers.dm`, namely in this section:
```
if(leaping && CanHug(hit_atom)) //Standard leaping behaviour, not attributable to being _thrown_ such as by a Carrier.
    Attach(hit_atom)
else if(hit_atom.density) //We hit something, cool.
    stat = UNCONSCIOUS //Giving it some brief downtime before jumping on someone via movement.
    icon_state = "[initial(icon_state)]_inactive"
    step(src, turn(dir, 180)) //We want the hugger to bounce off if it hits a mob.
    throwing = FALSE
    if(CanHug(hit_atom)) //We hit a host! Even cooler.
        spawn(5)
            if(loc && stat != DEAD)
                GoActive(5) //Up and at them! 0.5 second delay.
                icon_state = "[initial(icon_state)]"
    return
```
It accounts for no return to a CONSCIOUS state if it fails to attach.

**Comment by Ghommie** [comment]
#369 should be fixing it. Thanks.

**Comment by DominikPanic** [comment]
Is this fixed @Ghommie or does it require #451 ?

**Comment by Ghommie** [comment]
Requires the latter as it still goes perma-inactive in some edge cases.
